### DISCUSSION TITLE
Autoencoder and optuna

### DISCUSSION THREAD

**Discussion by unipdjahanifar** [optuna/optuna#3932] [Q&A]
### Don't use GitHub Issues to ask support questions.

Hi, I am trying to use Optuna and Autoencoder, but I face with the below error. Could you please let me know how I can solve the issue?

only batches of spatial targets supported (3D tensors) but got targets of size: : [256]

The Autoencoder is as below:

```python
class AutoEncoderOptuna(nn.Module):
    def __init__(self, trial):
        super().__init__()
        latent_space_dim = trial.suggest_int('latent_space_dim', 40, 80)
        num_chn = trial.suggest_int('n_channels', 8, 16)        
        self.encoder = nn.Sequential( 
                nn.Conv2d(1, num_chn, stride=(2, 2), kernel_size=(3, 3), padding=1), 
                nn.LeakyReLU(0.01),
                nn.Conv2d(num_chn, 2*num_chn, stride=(2, 2), kernel_size=(3, 3), padding=1), 
                nn.LeakyReLU(0.01),
                nn.Conv2d(2*num_chn, 4*num_chn, stride=(2, 2), kernel_size=(3, 3), padding=1), 
                nn.LeakyReLU(0.01),                
                nn.Flatten(), #16 * (4*num_chn) = 64*num_chn
                nn.Linear(64*num_chn, latent_space_dim),

        )
        self.decoder = nn.Sequential(
                torch.nn.Linear(latent_space_dim, 64*num_chn),
                Reshape(-1, 4*num_chn, 4, 4),
                nn.ConvTranspose2d(4*num_chn, 2*num_chn, stride=(2, 2), kernel_size=(3, 3), padding=1),
                nn.LeakyReLU(0.01),
                nn.ConvTranspose2d(2*num_chn, num_chn, stride=(2, 2), kernel_size=(3, 3), padding=1),                
                nn.LeakyReLU(0.01),
                nn.ConvTranspose2d(num_chn, 1, stride=(2, 2), kernel_size=(3, 3), padding=0),                
                nn.Sigmoid()
                )
        
        self.configure_loss(loss_func)

    def forward(self, x):
        x = self.encoder(x)
        x = self.decoder(x)
        return x


    def configure_loss(self, loss_func):
        self.loss_func = loss_func
```

**Comment by nzw0301** [comment]
This is not the error about optuna. The expected output shape doesn't match with the actual output to compute the loss according to the error message.
